stock_move_location/wizard/stock_move_location.py

```diff
@@ -75,6 +63,32 @@ def _compute_readonly_locations(self):
                 rec.origin_location_disable = True
                 rec.destination_location_disable = True
 
+    @api.depends("origin_location_id")
+    def _compute_picking_type_id(self):
+        company_id = self.env.context.get("company_id") or self.env.user.company_id.id
+        for rec in self:
+            picking_type = False
+            if rec.origin_location_id:
+                location_id = rec.origin_location_id
+                while location_id and not picking_type:
+                    domain = [
+                        ("code", "=", "internal"),
+                        ("warehouse_id.company_id", "=", company_id),
+                        ("default_location_src_id", "=", location_id.id),
+                    ]
+                    picking_type = self.env["stock.picking.type"].search(
+                        domain, limit=1
+                    )
+                    # Move up to the parent location if no picking type found
+                    location_id = not picking_type and location_id.location_id or False
+            if not picking_type:
+                domain = [
+                    ("code", "=", "internal"),
+                    ("warehouse_id.company_id", "=", company_id),
+                ]
+                picking_type = self.env["stock.picking.type"].search(domain, limit=1)
```

Comment:
Non blocking. Just to make it a bit shorter.
```suggestion
            picking_type = self.env["stock.picking.type"]
            base_domain = [
                ("code", "=", "internal"),
                ("warehouse_id.company_id", "=", company_id),
            ]
            if rec.origin_location_id:
                location_id = rec.origin_location_id
                while location_id and not picking_type:
                    domain = [("default_location_src_id", "=", location_id.id)]
                    domain = expression.AND([base_domain, domain])
                    picking_type = picking_type.search(domain, limit=1)
                    # Move up to the parent location if no picking type found
                    location_id = not picking_type and location_id.location_id or False
            if not picking_type:
                picking_type = picking_type.search(base_domain, limit=1)
```

Reply:
We cant use this `picking_type = self.env["stock.picking.type"]` because if the first `if` fails to assign picking_type, the search of second `if` will raise error.